proof/dleq.go

```diff
@@ -0,0 +1,108 @@
+package proof
+
+import (
+	"errors"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/hash"
+	"github.com/dedis/crypto/random"
+)
+
+// This package provides functionality to create and verify non-interactive
```

Comment:
https://blog.golang.org/godoc-documenting-go-code

Comments come _before_ the `package`, `struct`, `func`.